model/system_metrics.go

```diff
@@ -258,6 +275,42 @@ func (sm *SystemMetrics) appendSystemMetricsChunkKey(ctx context.Context, metric
 	return errors.Wrapf(err, "problem appending system metrics data chunk to %s", sm.ID)
 }
 
+// Download returns an io.Reader for the system metrics data of the specified type.
+func (sm *SystemMetrics) Download(ctx context.Context, metricType string) (*SystemMetricsReader, error) {
+	if sm.env == nil {
+		return nil, errors.New("cannot download system metrics with a nil environment")
+	}
+
+	if sm.ID == "" {
+		sm.ID = sm.Info.ID()
+	}
+
+	conf := &CedarConfig{}
+	conf.Setup(sm.env)
+	if err := conf.Find(); err != nil {
+		return nil, errors.Wrap(err, "problem getting application configuration")
+	}
+
+	bucket, err := sm.Artifact.Options.Type.Create(
+		ctx,
+		sm.env,
+		conf.Bucket.SystemMetricsBucket,
+		sm.Artifact.Prefix,
+		string(pail.S3PermissionsPrivate),
+		false,
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "problem creating bucket")
+	}
+
+	chunks, ok := sm.Artifact.MetricChunks[metricType]
+	if !ok {
+		return nil, fmt.Errorf("Invalid metric type %s for id %s", metricType, sm.ID)
+	}
+
+	return NewSystemMetricsReader(ctx, bucket, chunks, 2), nil
```

Comment:
Is there a reason the batch size is 2?

Reply:
based off the log download method, Julian said having two in parallel was good enough